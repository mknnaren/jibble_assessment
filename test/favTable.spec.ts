import { mount, createLocalVue  } from '@vue/test-utils'
import favTable from '@/components/favourites/table.vue';
import Vuetify from 'vuetify';
import Vue from 'vue'
Vue.use(Vuetify)

describe('favTable', () => {
  let localVue = createLocalVue()
  let vuetify = new Vuetify()


  test('mounts properly', () => {
    const wrapper = mount(favTable, {localVue, vuetify, propsData: {
      favPropList: [{
        imdbID: "ab123", Title: "Harry", favourite: true, Year: "2021" 
      }]}})
    expect(wrapper.vm).toBeTruthy()
  })
  test('mounts properly without prop', () => {
    const wrapper = mount(favTable, {localVue, vuetify
    })
    expect(wrapper.vm).toBeTruthy()
  })
  
  test('remove fav click event', async() => {
    const wrapper = mount(favTable, {localVue, vuetify, propsData: {
      favPropList: [{
        imdbID: "ab123", Title: "Harry", favourite: true, Year: "2021" 
      }]}})
    expect(wrapper.vm).toBeTruthy()
    const rmFav = wrapper.findAll(".rmFavDiv")
    await rmFav.at(0).trigger('click')
  })
  // test('renders properly', () => {
  //   const wrapper = mount(favTable,  {localVue, vuetify})
  //   expect(wrapper.html()).toMatchSnapshot()
  // })
})
