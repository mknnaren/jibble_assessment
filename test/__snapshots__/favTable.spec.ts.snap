// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favTable renders properly 1`] = `
"<div>
  <div class=\\"v-card v-sheet theme--light\\">
    <div class=\\"v-card__title\\">
      <div class=\\"v-input v-input--hide-details theme--light v-text-field v-text-field--single-line\\">
        <div class=\\"v-input__control\\">
          <div class=\\"v-input__slot\\">
            <div class=\\"v-text-field__slot\\"><label for=\\"input-58\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Search</label><input id=\\"input-58\\" type=\\"text\\"></div>
            <div class=\\"v-input__append-inner\\">
              <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-magnify theme--light\\"></i></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"v-data-table elevation-1 v-data-table--has-bottom theme--light\\">
      <div class=\\"v-data-table__wrapper\\">
        <table>
          <colgroup>
            <col class=\\"\\">
            <col class=\\"\\">
            <col class=\\"\\">
            <col class=\\"\\">
          </colgroup>
          <thead class=\\"v-data-table-header\\">
            <tr>
              <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Title\\" class=\\"text-left\\"><span>Title</span></th>
              <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Year\\" class=\\"text-start\\"><span>Year</span></th>
              <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"IMDB ID\\" class=\\"text-start\\"><span>IMDB ID</span></th>
              <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"\\" class=\\"text-start\\"><span></span></th>
            </tr>
          </thead>
          <tbody>
            <tr class=\\"v-data-table__empty-wrapper\\">
              <td colspan=\\"4\\">No data available</td>
            </tr>
          </tbody>
        </table>
      </div>
      <div class=\\"v-data-footer\\">
        <div class=\\"v-data-footer__select\\">Rows per page:<div class=\\"v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select\\">
            <div class=\\"v-input__control\\">
              <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-65\\" class=\\"v-input__slot\\">
                <div class=\\"v-select__slot\\">
                  <div class=\\"v-select__selections\\">
                    <div class=\\"v-select__selection v-select__selection--comma\\">10</div><input aria-label=\\"Rows per page:\\" id=\\"input-65\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\">
                  </div>
                  <div class=\\"v-input__append-inner\\">
                    <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                  </div><input type=\\"hidden\\" value=\\"10\\">
                </div>
                <div class=\\"v-menu\\">
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"v-data-footer__pagination\\">â€“</div>
        <div class=\\"v-data-footer__icons-before\\"><button type=\\"button\\" disabled=\\"disabled\\" class=\\"v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default\\" aria-label=\\"Previous page\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></span></button></div>
        <div class=\\"v-data-footer__icons-after\\"><button type=\\"button\\" disabled=\\"disabled\\" class=\\"v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default\\" aria-label=\\"Next page\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></span></button></div>
      </div>
    </div>
  </div>
</div>"
`;
